socketshark/receiver.py

```diff
@@ -24,21 +24,24 @@ def __init__(self, shark, redis_receiver):
         # {subscription: [sessions]}
         self.confirmed_subscriptions = defaultdict(set)
 
-        self.redis_settings = shark.config['REDIS']
-        self.redis_channel_prefix = self.redis_settings['channel_prefix']
-        self.redis = shark.redis
-        self.redis_receiver = redis_receiver
+        self.redis_connections = shark.redis_connections
 
         # We use a special channel to pass the stop message to the reader.
-        self._stop_channel = self.redis_receiver.channel('_internal')
+        self._stop_channel = self.redis_connections[0].redis_receiver.channel(
+            '_internal'
+        )
         self._stop = False  # Stop flag
```

Comment:
Can you elaborate on how this works and why we only need to have one stop channel for N Redis connections?

Reply:
Thanks for flagging this, I think I'm wrong here. I was assuming a single stop channel should work but that would only end one of the reader coroutines. Any others would continue to run and we'd still be waiting.

Reply:
I and updated a test to verify that the receiver does completely shut down. This did hang with a single stop channel. So I added a stop channel to each RedisConnection.

[09767cd](https://github.com/closeio/socketshark/pull/105/commits/09767cd33afab35c082987d179f6db7fbe57b003).